### PR TITLE
feat(modal): add backdrop and keyboard options

### LINKED ISSUE #134: Optionally hide modal backdrop in Modal
### Issue description
- While using Modal it would be nice if Modal has an option to hide the backdrop. Right now it is compulsorily shown and there is no way to configure the Modal to not show the backdrop.

Could be similar to [bootstrap v4 modal](http://v4-alpha.getbootstrap.com/components/modal/#options)
- version `3.1.0`
- components: `Modal`

### PR DESCRIPTION
**Pull request by TheSharpieOne**
Closes #134
Add backdrop prop to modal with the following options
 - `true` show backdrop, close on click (default)
 - `false` hide backdrop
 - `"static"` show backdrop, do not close on click

Add keyboard prop to modal with the following options
 - `true` close on escape (default)
 - `false` do not close on escape

Add tests and docs

### CODE REVIEW

File: src/TabContent.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ export default class TabContent extends Component {
     const classes = classnames('tab-content', this.props.className);
     return (
       <div className={classes}>
-        { this.props.children }
```
Comment (TheSharpieOne) [inline src/TabContent.js]: This was given me a linter error when running the tests.

**Comment by ajainarayanan** [comment]
@TheSharpieOne That was super quick! Thanks!

File: src/TabContent.js (hunk 1 above)
Comment (ajainarayanan) [inline src/TabContent.js]: For some reason it didn't throw me this error. Not sure if @eddywashere saw this when he release 3.1.0.

Comment (eddywashere) [inline src/TabContent.js]: missed it, maybe it needs option to exit properly in build step

**Comment by eddywashere** [comment]
great additions!

**Pull request merged**